error CS0012 while building on Linux

I am attempting to build this on Linux Mint (ubuntu derivative), but it fails with following message:

```
OBKDeviceAPI.cs(125,61): error CS0012: The type `System.Configuration.ConfigurationSection' is defined in an assembly that is not referenced. Consider adding a reference to assembly `System.Configuration, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

Here is the full output. Can someone help?

    carlig@mint ~/experiments/BK7231GUIFlashTool $ xbuild 
    
    >>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<
    
    XBuild Engine Version 14.0
    Mono, Version 6.8.0.105
    Copyright (C) 2005-2013 Various Mono authors
    
    Build started 6/9/2024 10:48:32 AM.
    __________________________________________________
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln:  warning : Don't know how to handle GlobalSection ExtensibilityGlobals, Ignoring.
    Project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln" (default target(s)):
	    Target ValidateSolutionConfiguration:
		    Building solution configuration "Debug|Any CPU".
	    Target Build:
		    Project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher/BK7231Flasher.csproj" (default target(s)):
			    Target PrepareForBuild:
				    Configuration: Debug Platform: AnyCPU
			    Target GenerateSatelliteAssemblies:
			    No input files were specified for target GenerateSatelliteAssemblies, skipping.
			    Target CoreCompile:
				    Tool /usr/lib/mono/4.5/mcs.exe execution started with arguments: /noconfig /debug:full /debug:portable /optimize- /out:obj/Debug/BK7231Flasher.exe /resource:obj/Debug/BK7231Flasher.FormDownloader.resources /resource:obj/Debug/BK7231Flasher.FormExtractedConfig.resources /resource:obj/Debug/BK7231Flasher.FormMain.resources /resource:obj/Debug/BK7231Flasher.FormOBKConfig.resources /resource:obj/Debug/BK7231Flasher.FormPin.resources /resource:obj/Debug/BK7231Flasher.FormCustom.resources /resource:obj/Debug/BK7231Flasher.FormPrompt.resources /resource:obj/Debug/BK7231Flasher.Properties.Resources.resources BitUtils.cs BK7231Flasher.cs BKFlashList.cs Colors.cs ConfigBase.cs CRC.cs FormDownloader.cs FormDownloader.Designer.cs FormExtractedConfig.cs FormExtractedConfig.Designer.cs FormMain.cs FormMain.Designer.cs FormMain_Net_Scanner.cs FormMain_Net_SingleDevice.cs FormMain_TuyaConfigExtractor.cs FormOBKConfig.cs FormOBKConfig.Designer.cs FormPin.cs FormPin.Designer.cs FormCustom.cs FormCustom.Designer.cs FormPrompt.cs FormPrompt.Designer.cs ILogListener.cs MiscUtils.cs ModifyProgressBarColor.cs MySettings.cs OBKConfig.cs OBKDeviceAPI.cs OBKFlags.cs OBKFlashLayout.cs OBKMassBackup.cs OBKRoles.cs OBKScanner.cs Program.cs Properties/AssemblyInfo.cs Rand.cs RequiredLibrariesCheck.cs TuyaConfig.cs RFPartitionUtil.cs RichTextUtil.cs SecurityProtocolTypeExtensions.cs TuyaModules.cs Properties/Resources.Designer.cs Properties/Settings.Designer.cs /target:winexe /win32icon:bk_icon.ico /define:"DEBUG;TRACE" /nostdlib /platform:AnyCPU /reference:../packages/Newtonsoft.Json.13.0.3/lib/net35/Newtonsoft.Json.dll /reference:/usr/lib/mono/2.0-api/System.dll /reference:/usr/lib/mono/2.0-api/System.Data.dll /reference:/usr/lib/mono/gac/System.Deployment/4.0.0.0__b03f5f7f11d50a3a/System.Deployment.dll /reference:/usr/lib/mono/2.0-api/System.Drawing.dll /reference:/usr/lib/mono/2.0-api/System.Net.dll /reference:/usr/lib/mono/2.0-api/System.Windows.Forms.dll /reference:/usr/lib/mono/2.0-api/System.Xml.dll /reference:/usr/lib/mono/2.0-api/System.Core.dll /reference:/usr/lib/mono/2.0-api//mscorlib.dll /warn:4
    BK7231Flasher.cs(403,17): warning CS0219: The variable `realRead' is assigned but its value is never used
    BK7231Flasher.cs(410,30): warning CS0168: The variable `ex' is declared but never used
    BK7231Flasher.cs(543,37): warning CS0168: The variable `ex' is declared but never used
    BK7231Flasher.cs(726,22): warning CS0219: The variable `val' is assigned but its value is never used
    BK7231Flasher.cs(973,24): warning CS0219: The variable `otherMode' is assigned but its value is never used
    FormMain.cs(172,17): warning CS0162: Unreachable code detected
    FormMain.cs(163,29): warning CS0168: The variable `ex' is declared but never used
    FormMain.cs(281,29): warning CS0168: The variable `ex' is declared but never used
    FormMain.cs(758,30): warning CS0168: The variable `ex' is declared but never used
    FormMain.cs(804,30): warning CS0168: The variable `ex' is declared but never used
    FormMain.cs(884,13): warning CS0162: Unreachable code detected
    FormMain.cs(899,13): warning CS0162: Unreachable code detected
    FormMain.cs(1005,29): warning CS0168: The variable `ex' is declared but never used
    FormMain.cs(1239,29): warning CS0219: The variable `savedI' is assigned but its value is never used
    FormMain.cs(1331,30): warning CS0168: The variable `ex' is declared but never used
    FormOBKConfig.cs(44,30): warning CS0168: The variable `ex' is declared but never used
    FormOBKConfig.cs(171,21): warning CS0162: Unreachable code detected
    FormOBKConfig.cs(336,29): warning CS0168: The variable `ex' is declared but never used
    FormPin.cs(46,29): warning CS0168: The variable `ex' is declared but never used
    FormCustom.cs(49,30): warning CS0168: The variable `ex' is declared but never used
    OBKDeviceAPI.cs(125,61): error CS0012: The type `System.Configuration.ConfigurationSection' is defined in an assembly that is not referenced. Consider adding a reference to assembly `System.Configuration, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
				    /usr/lib/mono/2.0-api/System.dll (Location of the symbol related to previous error)
			    Task "Csc" execution -- FAILED
			    Done building target "CoreCompile" in project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher/BK7231Flasher.csproj".-- FAILED
		    Done building project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher/BK7231Flasher.csproj".-- FAILED
	    Task "MSBuild" execution -- FAILED
	    Done building target "Build" in project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln".-- FAILED
    Done building project "/home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln".-- FAILED
    
    Build FAILED.
    
    Warnings:
    
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln:  warning : Don't know how to handle GlobalSection ExtensibilityGlobals, Ignoring.
    
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln (default targets) ->
    (Build target) ->
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher/BK7231Flasher.csproj (default targets) ->
    /usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->
    
	    BK7231Flasher.cs(403,17): warning CS0219: The variable `realRead' is assigned but its value is never used
	    BK7231Flasher.cs(410,30): warning CS0168: The variable `ex' is declared but never used
	    BK7231Flasher.cs(543,37): warning CS0168: The variable `ex' is declared but never used
	    BK7231Flasher.cs(726,22): warning CS0219: The variable `val' is assigned but its value is never used
	    BK7231Flasher.cs(973,24): warning CS0219: The variable `otherMode' is assigned but its value is never used
	    FormMain.cs(172,17): warning CS0162: Unreachable code detected
	    FormMain.cs(163,29): warning CS0168: The variable `ex' is declared but never used
	    FormMain.cs(281,29): warning CS0168: The variable `ex' is declared but never used
	    FormMain.cs(758,30): warning CS0168: The variable `ex' is declared but never used
	    FormMain.cs(804,30): warning CS0168: The variable `ex' is declared but never used
	    FormMain.cs(884,13): warning CS0162: Unreachable code detected
	    FormMain.cs(899,13): warning CS0162: Unreachable code detected
	    FormMain.cs(1005,29): warning CS0168: The variable `ex' is declared but never used
	    FormMain.cs(1239,29): warning CS0219: The variable `savedI' is assigned but its value is never used
	    FormMain.cs(1331,30): warning CS0168: The variable `ex' is declared but never used
	    FormOBKConfig.cs(44,30): warning CS0168: The variable `ex' is declared but never used
	    FormOBKConfig.cs(171,21): warning CS0162: Unreachable code detected
	    FormOBKConfig.cs(336,29): warning CS0168: The variable `ex' is declared but never used
	    FormPin.cs(46,29): warning CS0168: The variable `ex' is declared but never used
	    FormCustom.cs(49,30): warning CS0168: The variable `ex' is declared but never used
    
    Errors:
    
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher.sln (default targets) ->
    (Build target) ->
    /home/carlig/experiments/BK7231GUIFlashTool/BK7231Flasher/BK7231Flasher.csproj (default targets) ->
    /usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->
    
	    OBKDeviceAPI.cs(125,61): error CS0012: The type `System.Configuration.ConfigurationSection' is defined in an assembly that is not referenced. Consider adding a reference to assembly `System.Configuration, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
    
	     21 Warning(s)
	     1 Error(s)
    
    Time Elapsed 00:00:01.1749590
    carlig@mint ~/experiments/BK7231GUIFlashTool $ ^C
    carlig@mint ~/experiments/BK7231GUIFlashTool $

--- a/BK7231Flasher/BK7231Flasher.csproj
+++ b/BK7231Flasher/BK7231Flasher.csproj
@@ -46,6 +46,7 @@
     <Reference Include="System.Net" />
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xml" />
+    <Reference Include="System.Configuration" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="BitUtils.cs" />

